tests/visualization/test_graphviz.py

```diff
@@ -150,3 +151,50 @@ def test_filename(self):
         self.assertTrue(os.path.isfile("test_graphviz_filename.svg"))
         if not SAVE_IMAGES:
             self.addCleanup(os.remove, "test_graphviz_filename.svg")
+
+    def test_escape_sequences(self):
+        # Create a simple graph
+        graph = rustworkx.generators.path_graph(2)
+
+        escapes = [
+                    "\n",  # Newline
+                    "\t",  # Horizontal tab
+                    "\'",  # Single quote
+                    "\"",  # Double quote
+                    "\\",  # Backslash
+                    "\r",  # Carriage return
+                    "\b",  # Backspace
+                    "\f",  # Form feed
+                ]
+
+        def node_attr(node):
+            """
+            Define node attributes including randomly inserted escape sequences
+            """
+
+            base_label = "label"
+            base_tooltip = "tooltip"
+            insert_at_label = random.randint(0, len(base_label))
+            insert_at_tooltip = random.randint(0, len(base_tooltip))
+            label = ''.join([
+                                base_label[:insert_at_label],
+                                random.choice(escapes),
+                                base_label[insert_at_label:]
+                            ])
+            tooltip = ''.join([
+                                base_tooltip[:insert_at_tooltip],
+                                random.choice(escapes),
+                                base_tooltip[insert_at_tooltip:]
+                            ])
+            return {"label": label, "tooltip": tooltip}
+
+        # Draw the graph using graphviz_draw
+        dot_str = graph.to_dot(node_attr)
+        dot = pydot.graph_from_dot_data(dot_str)[0]
+
+        with tempfile.TemporaryDirectory() as tmpdirname:
```

Comment:
This is exactly what you should avoid. Instead of writing the image, just check if the string from `to_dot` contains the escaped characters

Reply:
Got it, Thank you for the clarification

Reply:
I have updated the code to check the to_dot for escape sequence, also looped through all possible ones to ensure we don't miss any